src/engine/src/registry/Registry.cpp

```diff
@@ -3,35 +3,20 @@
 ES::Engine::Registry::Registry() : _registry(nullptr)
 {
     this->_registry = std::make_unique<entt::registry>();
-    this->_systems = std::unordered_map<ScheduleLabel, std::vector<USystem>>();
-    this->_systems[ScheduleLabel::NON_FIXED] = std::vector<USystem>();
-    this->_systems[ScheduleLabel::FIXED] = std::vector<USystem>();
-    this->_fixedUpdateclock = ES::Utils::Clock();
+
+    this->RegisterScheduler<ES::Engine::Scheduler::Startup>();
+    this->RegisterScheduler<ES::Engine::Scheduler::Update>();
+    this->RegisterScheduler<ES::Engine::Scheduler::FixedTimeUpdate>();
+    this->RegisterScheduler<ES::Engine::Scheduler::RelativeTimeUpdate>();

```

Comment:
Btw, this will allow to remove:
```cpp
namespace ES::Engine {
class Registry;
}
```
when you're creating those schedulers

Reply:
I agree with the fact that the RealTimeProvider resource should be deleted (as it is now useless).
For the "EmptyRegistry" I need more clarification. Are you thinking about an interface / abstract or just an "EmptyRegistry" that has no schedulers ? (the latter option being very confusing I think)

Reply:
I will also remove the "clock" utils as well as its logic is now implemented directly into the "FixedTimeScheduler"

Reply:
Actually I have to implement the EmptyRegistry to allow for scheduler deletion so I will do it

Reply:
I fixed it using callbacks because the EmptyRegistry was messing everything, let me know how you find its implementation

Reply:
>Are you thinking about an interface / abstract or just an "EmptyRegistry" that has no schedulers ? (the latter option being very confusing I think)

As we have template inside of Registry, we can't create an interface and an abstract shouldn't be really useful. I was thinking about a registry that doesn't have scheduler registered at all. Why do you think "the latter option being very confusing I think" ? It's because user doesn't know if we talk about entity(component) or systems and schedulers ? In that case, namming is the problem but exept putting documentation, I don't have a better name for it :/

Reply:
> I will also remove the "clock" utils as well as its logic is now implemented directly into the "FixedTimeScheduler"

Rather than removing it, I think you should create a TickClock or something like that: we gonna use clock for different purpose later like if we want to create threads.

Reply:
> I fixed it using callbacks because the EmptyRegistry was messing everything, let me know how you find its implementation

It's impracticable to use declared `Registry` because we can't call methods of it. So rather than using a callback, calling `DeleteScheduler` should be cleaner.

Reply:
we can't

Reply:
Please split the different topics of discussion as comments to solve separately @Miou-zora becaues this conversation is getting too big and we are talking about too much stuff in one place, I am lost 😂

Reply:
As we talked about it in meeting, we can't create EmptyRegistry for technical purpose and TickClock would not be useful at the moment.